testutils/clustermanager/gke.go

```diff
@@ -123,55 +145,46 @@ func (gsc *GKESDKClient) setAutoscaling(project, clusterName, location, nodepool
 	return gsc.Service.Projects.Locations.Clusters.NodePools.SetAutoscaling(parent, rb).Do()
 }
 
-// Setup sets up a GKECluster client.
-// minNodes: default to 1 if not provided
-// maxNodes: default to 3 if not provided
-// nodeType: default to n1-standard-4 if not provided
-// region: default to regional cluster if not provided, and use default backup regions
-// zone: default is none, must be provided together with region
-// project: no default
-// addons: cluster addons to be added to cluster
-func (gs *GKEClient) Setup(minNodes *int64, maxNodes *int64, nodeType *string, region *string, zone *string, project *string, addons []string) ClusterOperations {
-	gc := &GKECluster{
-		Request: &GKERequest{
-			MinNodes:      DefaultGKEMinNodes,
-			MaxNodes:      DefaultGKEMaxNodes,
-			NodeType:      DefaultGKENodeType,
-			Region:        DefaultGKERegion,
-			Zone:          DefaultGKEZone,
-			BackupRegions: DefaultGKEBackupRegions,
-			Addons:        addons,
-		},
-	}
+// Setup sets up a GKECluster client, takes GEKRequst as parameter and applies
+// all defaults if not defined.
+func (gs *GKEClient) Setup(r GKERequest) ClusterOperations {
+	gc := &GKECluster{}
 
-	if project != nil { // use provided project and create cluster
-		gc.Project = project
+	if r.Project != "" { // use provided project and create cluster
+		gc.Project = &r.Project
 		gc.NeedCleanup = true
 	}
 
-	if minNodes != nil {
-		gc.Request.MinNodes = *minNodes
-	}
-	if maxNodes != nil {
-		gc.Request.MaxNodes = *maxNodes
+	if r.MinNodes == 0 {
```

Comment:
There can be a chance that user sets `MinNodes` as a value that is >3, but does not set `MaxNodes`, then the cluster creation will result in an error. How about setting `MaxNodes` to be equal to `MinNodes` for such case, or adding some kind of verification?

Reply:
Also the document says `--min-nodes` and `--max-nodes` will be ignored if autoscaling is disabled, see https://cloud.google.com/sdk/gcloud/reference/container/clusters/create#. Though it's for `gcloud` command, I think it's the same for the SDK.
Do we have autoscaling enabled by default?

Reply:
yes we have autoscaling enabled by default. It's a good idea to avoid only set minNodes

Reply:
done